api/views/phones.py

```diff
@@ -640,6 +643,168 @@ def sms_status(request):
     return response.Response(status=200)
 
 
+call_body = openapi.Schema(
+    type=openapi.TYPE_OBJECT,
+    required=["to"],
+    properties={"to": openapi.Schema(type=openapi.TYPE_STRING)},
+)
+
+
+@decorators.permission_classes([permissions.IsAuthenticated, HasPhoneService])
+@swagger_auto_schema(method="post", request_body=call_body)
+@decorators.api_view(["POST"])
+def call(request):
+    """
+    Make a call from the authenticated user's relay number.
+
+    """
+    if not flag_is_active(request, "outbound_phone"):
+        return response.Response({"outbound_phone": False}, status=404)
+    to = request.data.get("to", None)
+    if to is None:
+        raise exceptions.ValidationError("Missing 'to' parameter.")
+    real_phone = RealPhone.objects.get(user=request.user)
+    relay_number = RelayNumber.objects.get(user=request.user)
+    client = twilio_client()
+    client.calls.create(
+        twiml=f"<Response><Say>Dialing {to} ...</Say><Dial>{to}</Dial></Response>",
+        to=real_phone.number,
+        from_=relay_number.number,
+    )
+    return response.Response(status=200)
+
+
+message_body = openapi.Schema(
+    type=openapi.TYPE_OBJECT,
+    properties={
+        "body": openapi.Schema(type=openapi.TYPE_STRING),
+        "destination": openapi.Schema(type=openapi.TYPE_STRING),
+    },
+)
+
+
+@decorators.permission_classes([permissions.IsAuthenticated, HasPhoneService])
+@swagger_auto_schema(method="post", request_body=message_body)
+@decorators.api_view(["POST"])
+def post_message(request):
+    """
+    Send a message from the user's realy number.
+
+    POST params:
+        body: the body of the message
+        destination: E.164-formatted phone number
+
+    """
+    if not flag_is_active(request, "outbound_phone"):
+        return response.Response({"outbound_phone": False}, status=404)
+    relay_number = RelayNumber.objects.get(user=request.user)
+    body = request.data.get("body")
+    destination_number = request.data.get("destination")
+    client = twilio_client()
+    client.messages.create(from_=relay_number.number, body=body, to=destination_number)
+    return response.Response(
+        status=200,
+    )
+
+
+messages_body = openapi.Schema(
+    type=openapi.TYPE_OBJECT,
+    properties={"with": openapi.Schema(type=openapi.TYPE_STRING)},
+)
```

Comment:
The `@swagger_auto_schema` is used on these views without a serializer to document the parameters you can pass to the function. They are displayed on `/api/v1/docs`.  See https://drf-yasg.readthedocs.io/en/stable/custom_spec.html#the-swagger-auto-schema-decorator

There are decorators on the `/api/v1/call` and `/api/v1/message` endpoints, but not on `/api/v1/messsages`. It looks like this was a work in progress, since the way to document `GET` query string parameters is different than the `POST` body. I've updated it to `messages_query` and added the `@swagger_auto_schema` to the method `list_messages` implementing `/api/v1/messages`.